src/colorizer/CanvasUIOverlay.ts

```diff
@@ -88,20 +82,35 @@ export default class CanvasOverlay {
     timestampOptions: TimestampOptions = defaultTimestampOptions,
     overlayOptions: OverlayFillOptions = defaultBackgroundOptions
   ) {
-    this.canvas = new OffscreenCanvas(1, 1);
+    super();
+
+    this.canvas = document.createElement("canvas");
+    this.canvas.style.display = "block";
+
     this.scaleBarOptions = scaleBarOptions;
     this.timestampOptions = timestampOptions;
     this.backgroundOptions = overlayOptions;
     this.canvasWidth = 1;
     this.canvasHeight = 1;
   }
 
-  /**
-   * Set the size of the canvas overlay.
-   */
-  setSize(width: number, height: number): void {
+  // Wrapped ColorizeCanvas functions ///////
+
+  get domElement(): HTMLCanvasElement {
+    // Override base ColorizeCanvas getter with the composited canvas.
+    return this.canvas;
+  }
+
+  public setSize(width: number, height: number): void {
     this.canvasWidth = width;
     this.canvasHeight = height;
+    super.setSize(width, height);
+  }
+
+  // Rendering ////////////////////////////////
+
+  private getPixelRatio(): number {
```

Comment:
Refactoring the pixel ratio logic into a method really makes the code cleaner and more modular!